### PR TITLE
feat: expose secondary_storage

### PR DESCRIPTION
**Pull request by jor2**
### Description

Surface the secondary storage field https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/container_vpc_worker_pool#secondary_storage in landing zone interface (via override.json).

https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone/issues/688

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Surface the secondary storage field https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/container_vpc_worker_pool#secondary_storage in landing zone interface (via override.json).

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
@jor2 FYI pipeline is blocked due to a backend bug - we are working on a workaround to ignore the error in our tests to unblock PRs as backend fix won't be in prod for 2-3 weeks

File: patterns/roks/module/variables.tf [hunk 1]
```diff
@@ -163,6 +163,12 @@ variable "entitlement" {
   default     = null
 }
 
+variable "secondary_storage" {
+  description = "Secondary storage type for cluster. Use the IBM Cloud CLI command `ibmcloud ks flavors` to find valid machine types, e.g `ibmcloud ks flavor get --flavor bx2.16x64 --provider vpc-gen2 --zone us-south-1`"
```
Comment (ocofaigh) [inline patterns/roks/module/variables.tf:167]: Optionally specify a secondary storage option to attach to all cluster worker nodes. This value is immutable and can't be changed after provisioning. Use the IBM Cloud CLI command `ibmcloud ks flavors` to find valid options, e.g `ibmcloud ks flavor get --flavor bx2.16x64 --provider vpc-gen2 --zone us-south-1`.

File: patterns/mixed/variables.tf [hunk 2]
```diff
@@ -177,6 +177,12 @@ variable "flavor" {
   default     = "bx2.16x64"
 }
 
+variable "secondary_storage" {
+  description = "Secondary storage type for cluster. Use the IBM Cloud CLI command `ibmcloud ks flavors` to find valid machine types, e.g `ibmcloud ks flavor get --flavor bx2.16x64 --provider vpc-gen2 --zone us-south-1`"
```
Comment (ocofaigh) [inline patterns/mixed/variables.tf:181]: Optionally specify a secondary storage option to attach to all cluster worker nodes. This value is immutable and can't be changed after provisioning. Use the IBM Cloud CLI command `ibmcloud ks flavors` to find valid options, e.g `ibmcloud ks flavor get --flavor bx2.16x64 --provider vpc-gen2 --zone us-south-1`.

File: examples/override-example/override.json [hunk 3]
```diff
@@ -8,8 +8,46 @@
...
+                    "vsi-zone-3"
+                ],
+                "vpc_name": "workload",
+                "workers_per_subnet": 2
+            }
+        ],
+        "workers_per_subnet": 2
+    }
+  ],
+"enable_transit_gateway": true,
```
Comment (ocofaigh) [inline examples/override-example/override.json:50]: indentation looks wrong here?

File: examples/override-example/override.json [hunk 4]
```diff
@@ -8,8 +8,46 @@
     "use_appid": false,
     "use_data": false
   },
-  "clusters": [],
-  "enable_transit_gateway": true,
+  "clusters": [
```
Comment (ocofaigh) [inline examples/override-example/override.json:11]: you should update this example readme since it now looks like it will deploy a cluster

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@jor2 I left some comments. BTW, is there no test for the override json example? I see the test `TestRunOverride` is actually running the VSI QS pattern and I wonder is that a mistake?

File: tests/pr_test.go [hunk 5]
```diff
@@ -23,6 +23,7 @@ import (
 	"github.com/terraform-ibm-modules/ibmcloud-terratest-wrapper/testhelper"
 )
 
+const overrideExampleTerraformDir = "examples/override-example"
```
[the remark this answers is not in the document]
Comment (jor2) [reply tests/pr_test.go:26]: I was going to add an override test, but there seems to be whole lot of tests running and wasn't sure if we wanted to add more.

[the remark this answers is not in the document]
Comment (ocofaigh) [reply tests/pr_test.go:26]: Every example should have a test - so we need to decide if the test is in pr_test.go, or other_test.go. Suggest you create a follow up issue to track the test creation. Keep it in a separate PR to this.

**Pull request merged**